plugin/src/main/java/io/jenkins/plugins/coverage/model/CodeDeltaCalculator.java

```diff
@@ -0,0 +1,111 @@
+            String previousCommit = previousBuildAction.lastBuild.getRevision().getSha1String();
+
+            DeltaCalculator deltaCalculator = getDeltaCalculator();
+            Optional<Delta> delta = deltaCalculator.calculateDelta(commit, previousCommit, log);
+
+            delta.ifPresent(value -> setChangedFilesInCoverageTree(rootNode, value));
+        }
+    }
+
+    private DeltaCalculator getDeltaCalculator() {
+        FilteredLog log = new FilteredLog("Calculate code delta");
+        SCM scm = ((AbstractBuild<?, ?>) build).getProject().getScm();
+        if (scm instanceof GitSCM) {
+            GitDeltaCalculatorFactory gitDeltaCalculatorFactory = new GitDeltaCalculatorFactory();
+            Optional<DeltaCalculator> gitDeltaCalculator = gitDeltaCalculatorFactory.createDeltaCalculator(scm, build,
+                    workspace, listener, log);
+            if (gitDeltaCalculator.isPresent()) {
+                return gitDeltaCalculator.get();
+            }
+        }
+        return DeltaCalculatorFactory
+                .findDeltaCalculator(build, Collections.singleton(workspace), listener, log);
+    }
+
+    private void setChangedFilesInCoverageTree(final CoverageNode coverageNode, final Delta delta) {
+        Map<String, CoverageNode> nodePathMapping = coverageNode.getAll(CoverageMetric.FILE).stream()
+                .map(node -> (FileCoverageNode) node)
+                .collect(Collectors.toMap(FileCoverageNode::getPath, Function.identity()));
+
+        List<FileChanges> fileChanges = delta.getFileChangesMap().values().stream()
+                .filter(fileChange -> fileChange.getFileEditType().equals(FileEditType.MODIFY)
+                        || fileChange.getFileEditType().equals(FileEditType.ADD))
+                .collect(Collectors.toList());
+
+        for (FileChanges fileChange : fileChanges) {
+            // remove maven default folders if existent since they are not required
+            String path = fileChange.getFileName()
+                    .replace(Paths.get("src", "main", "java") + File.separator, "");
+            if (nodePathMapping.containsKey(path)) {
+                CoverageNode changedNode = nodePathMapping.get(path);
+                if (changedNode.getMetric().equals(CoverageMetric.FILE)) {
+                    attachChangedCodeLines((FileCoverageNode) changedNode, fileChange);
+                }
+            }
+        }
+    }
+
+    private void attachChangedCodeLines(final FileCoverageNode changedNode, final FileChanges fileChange) {
+        Set<Change> relevantChanges = new HashSet<>();
+        relevantChanges.addAll(fileChange.getChangesByType(ChangeEditType.INSERT));
+        relevantChanges.addAll(fileChange.getChangesByType(ChangeEditType.REPLACE));
+        for (Change change : relevantChanges) {
+            for (int i = change.getFromLine(); i <= change.getToLine(); i++) {
+                changedNode.addChangedCodeLines(i);
+            }
```

Comment:
wenn addChangedCodeLines mit nur einer Zahl aufgerufen werden wäre eine addChangedCodeLine(int)-Methode besser, die dann nicht unnötig eine Zahl beim Aufruf in ein Array und eine List verpackt.